Tranche 2 (#187)
SpinalHDL/SpinalDoc-RTD: source/SpinalHDL/Data types/bits.rst

.. code-block:: scala

   // Declaration

>    val myBits  = Bits()     // the size is inferred through later assignment

Reviewer: I should have been clearer: The width is inferred from the assignments (actually the biggest known width) - so 6 bit width is inferred in your example.
But you try to assign a 4 bit value to a (inferred size) 6 bit value, which generates an error.

>    val myBits  = Bits()     // the size is inferred through later assignment

Author: So we ended up as:
```
   // Declaration
   val myBits  = Bits()     // the size is inferred from the widest assignment
   // ....
   // .resized needed to prevent WIDTH MISMATCH error as natural width does not match inferred size
   myBits := B("1010").resized  // auto-widen Bits(4 bits) to Bits(6 bits)
   when(condxMaybe) {
     myBits := B("110000")      // Bits(6 bits) would be inferred for myBits, this is the widest assignment
   }
```

This does help clarify the inference of hardware matter from the inference of Scala matters, which is the usual use of the term.

.. code-block:: scala

   // Declaration
   val myBits  = Bits()     // the size is inferred at first assignment
   // ....
   myBits := B("001010")    // Bits(6 bits) would be inferred
   when(condxMaybe) {
     myBits := B("110000")
   } elsewhen(condxMaybeTwo) {

>      //myBits := B("00110000")	// Would Error: WIDTH MISMATCH (8 bits <- 6 bits)

Reviewer: ```suggestion
     //myBits := B("0011")	// Would Error: we can't assign a 4 bit Bits
     to a 6 bit Bits w/o resizing
```
We should use your original example with 4 bits as it does not change which size is inferred - easier to understand IMO

   * - x.andR
     - AND all bits of x
     - Bool

>    * - | y = 1 // int

Author: This is up there now... https://dlmiles.github.io/SpinalDoc-RTD/docs_multi/html/vTEST/SpinalHDL/Data%20types/bits.html#logic  apart from the extra whitespace towards bottom of cell I think it look better.

I have fixed the use of `int` instead of `Int` to draw attention to types involved.

>    * - | y = 1 // int

Reviewer: While the newlines are now fixed, to some other things remain:
- The table entries now take up 3 lines (2 lines of content, one empty line)
- "int" is not a type, the correct scala type is "Int"
- I still think it makes it less readable to restate all information twice, both the type of "y" and the size of the output result are present

       | x \>\> y
     - | Logical shift right, y: Int

>        | Result may reduce width

Author: This is up there now... https://dlmiles.github.io/SpinalDoc-RTD/docs_multi/html/vTEST/SpinalHDL/Data%20types/bits.html#logic apart from the extra whitespace towards bottom of cell I think it look better.

The commit you are looking at looks like contents of the page up there now.

The Vertical Space is a CSS matter I can probably resolve later, not really something I would block the new content / commit on.

   * - x.andR
     - AND all bits of x
     - Bool

>    * - | y = 1 // Int

Reviewer: I still think that it's easier to read without the duplicated information.

Author: I think you are citing the TYPE info column as duplicate information ?  Maybe to you as an advanced user, but to a novice that is type information (like the heading says - I'm not suggesting a change here BTW).

It is much better to restate things as key-points in clear readable language than use a new foreign cryptic notation to the reader, that add additional load to the reader trying to decode/understand what is being read.

Maybe the right hand column would work well with a tooltip style hover over, explaining the meaning of the notation in plain language.

`The result is the same data type as the original at the same width.`

It would however remove it from the page for any PDF/screen-capture.

     - Return the index of the MSB (highest allowed index for Bits)
     - Int
   * - x.msb

>      - Return the most significant bit value

Reviewer: I find this one to be more confusing with the change, "most significant bit" is a well defined term, if you add "value" I get suspicious...

Reviewer: Interesing: to me, if you ask for the "the most significant bit" you ask for the bit, not it's position. - probably an indication that we should make this clear.

I'm a bit rusty on my english and not sure if that should then be "the most significant bit value" or "the most significant bits value", maybe go for "the value of the most significant bit"?

     - Bool
   * - x.lsb
     - Return the least significant bit value
     - Bool
   * - x.reverse

>      - Return a representation of the same instance in reverse bit order,

Reviewer: ```suggestion
     - Return a copy of x with reversed bit order
```
Use the same wording as below?

Author: I see what you mean about the use of the word `copy` in the item below or so.  I would argue this is also a bad choice and the word `representation` is a better word there as well, relating to a `data representation` of the same underlying thing.  In reading the docs I was trying not to be too nit-picky over such things.  But if I were I would change that too and use a distinct language word less over-used than the word copy (can mean a lot of different things, so lots of room for misunderstanding).

Reviewer: I had a quick look at the code: it looks to me like this should actually be `reversed` - overlooked that initially
https://github.com/SpinalHDL/SpinalHDL/blob/af55caaf3e9821eed2b2ad2d5bfda742716bcb43/core/src/main/scala/spinal/core/Bits.scala#L237
I used copy since the reversed vector will actually not let you modify the original one, i.e.
```scala
val x = Bits(5 bit)
val y = x.reversed
when(x(0)) { y(3) := True }
```
generates
```verilog
y = {x[0],{x[1],{x[2],{x[3],{x[4]}}}}}
if(...) begin
  y[3] = 1'b1;
end
```

     - Subdivide x in multiple slices of y bits, y: Int
     - Vec(Bits, w(x)/y)
   * - x.resize(y)

>      - Return a resized copy of x, if enlarged, it is filled with zero

Reviewer: ```suggestion
     - Return a resized copy of x, if enlarged, it is extended with zero
```
